src/lib/urn.ts

```diff
@@ -1,25 +1,106 @@
 import { getURNProtocol, Network } from '@dcl/schemas'
 import { getChainIdByNetwork } from 'decentraland-dapps/dist/lib/eth'
 
-export const VERSION = 1
-export const DELIMITER = ':'
+const VERSION = 1
 
-export function buildItemURN(type: string, name: string, description: string, category: string, bodyShapeTypes: string) {
+/**
+ * urn:decentraland:
+ *   (?<protocol>
+ *     mainnet|
+ *     ropsten|
+ *     matic|
+ *     mumbai|
+ *     off-chain
+ *   ):
+ *   (
+ *     (?<type>
+ *       base-avatars|
+ *       collections-v2|
+ *       collections-thirdparty
+ *     ):
+ *     (?<suffix>
+ *       ((?<=base-avatars:)BaseMale|BaseFemale)|
+ *       ((?<=collections-v2)0x[a-fA-F0-9]{40})|
+ *       ((?<=collections-thirdparty:)
+ *          (?<thirdPartyName>[^:|\\s]+)
+ *          (:(?<thirdPartyCollectionId>[^:|\\s]+))?
+ *          (:(?<thirdPartyTokenId>[^:|\\s]+))?
+ *       )
+ *     )
+ *   )
+ */
+const baseMatcher = 'urn:decentraland'
+const protocolMatcher = '(?<protocol>mainnet|ropsten|matic|mumbai|off-chain)'
+const typeMatcher = '(?<type>base-avatars|collections-v2|collections-thirdparty)'
+
+const baseAvatarsSuffixMatcher = '((?<=base-avatars:)BaseMale|BaseFemale)'
+const collectionsSuffixMatcher = '((?<=collections-v2:)0x[a-fA-F0-9]{40})'
+const thirdPartySuffixMatcher =
+  '((?<=collections-thirdparty:)(?<thirdPartyName>[^:|\\s]+)(:(?<thirdPartyCollectionId>[^:|\\s]+))?(:(?<thirdPartyTokenId>[^:|\\s]+))?)'
```

Comment:
Regex capture groups are around 3 year old and they don't have polyfills, do you know by any change which browser versions should we support?

Reply:
Checked when developing and found this:
![image](https://user-images.githubusercontent.com/692077/140194652-a5f9481d-bbf9-42b2-9573-ed5e90d6fef0.png)

https://caniuse.com/?search=regex%20capture

2020 for edge and FF, but saw `Global 93.2%` and went ahead with it